src/PlatformCompatibilityTests/ServiceControlCompatibilityTests/Infrastructure/Sql/SqlScTest.cs

```diff
@@ -1,33 +1,41 @@
 ﻿namespace ServiceControlCompatibilityTests
 {
     using NUnit.Framework;
+    using System;
+    using System.Collections.Generic;
 
     abstract class SqlScTest
     {
-        ServiceControlInstance serviceControl;
-
-        [SetUp]
-        public void Setup()
+        Dictionary<Type, Func<ITransportDetails>> transportDetailActivations = new Dictionary<Type, Func<ITransportDetails>>
         {
-            var context = TestContext.CurrentContext;
+            { typeof(SqlTransportDetails), () => new SqlTransportDetails("Data Source=.\\SQLEXPRESS;Initial Catalog=nservicebus;Integrated Security=True") }
+        };
 
-            // TODO: This is a build artifact from the latest SC build. It should be imported into the working folder of the build
-            var factory = new ServiceControlFactory(@"C:\Temp\ServiceControl");
+        [TearDown]
+        public void TearDown()
+        {
+            // TODO: Use transport to Clean Up
 
-            // TODO: ServiceControl SQL Broker Connection String
-            var transport = new SqlTransportDetails("Data Source=.\\SQLEXPRESS;Initial Catalog=nservicebus;Integrated Security=True");
+            serviceControl?.Stop();
+        }
 
-            // TODO: Use transport to Clean Up
+        protected ITransportDetails ActivateInstanceOfTransportDetail(Type transportDetailType)
+        {
+            Func<ITransportDetails> activator;
+            transportDetailActivations.TryGetValue(transportDetailType, out activator);
```

Comment:
Why TryGetValue? Wouldn't there always be a key for the configured datasource? Should we Assert.Fail if the type is not configured?